File: 066_Plus_One.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        size = len(digits)
        
        for i in range(size-1, -1, -1) :
            if digits[i] + 1 < 10:
                digits[i] += 1
                break
            digits[i] = 0
            if i == 0:
                digits.insert(0, 1)
        return digits

File: 066_Plus_One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        
        int size = digits.size();
        
        for (int i=size-1; i>=0; i--) {
            digits[i] += 1;
            if (digits[i] < 10)
                break;
            digits[i] = 0;
            if (i==0)
                digits.insert(digits.begin(), 1);
        }
        return digits;
    }
};